packages/cli-repl/src/mongosh-repl.ts

```diff
@@ -391,6 +394,31 @@ class MongoshNodeRepl implements EvaluationListener {
     return { __initialized: 'yes' };
   }
 
+  setupHistoryCommand(): void {
+    const history = () => {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const replHistory: string[] = (this.runtimeState().repl as any).history;
+      const formattedHistory =
+        // Remove the history call from the formatted history
+        replHistory.slice(1).reverse();
+
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      formattedHistory[util.inspect.custom as any] = (() => {
+        return formatOutput(
+          {
+            // Providing a copy of the history avoids a circular reference.
+            value: formattedHistory.concat(),
```

Comment:
Doesn't `.slice().reverse()` above already copy?

Reply:
it copies it for the sake of not mutating the original `history` but needs to be copied again

Reply:
Ah wait, this is because the `formattedHistory` value here has the custom inspect function?

I think I was misled by the comment here a bit, we're not avoiding a circular reference but rather infinite recursion, right?

Reply:
yeah, I thought of it as a circular reference since the inspect ends up (I'm assuming) looking into the given object's [util.inspect.custom] as well and then finding the original function again. But yeah it's meant to avoid infinite recursion either way, I'll reword that.